File: benchmarks/oopsla25/subjects/490.py
import sys

def f490(s: str):
  i = 0
  while i < len(s):
    assert s[i] == "0" or s[i] == "1"
    i += 1  
  assert s[-3:] == "011"

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f490(my_string)

File: benchmarks/oopsla25/subjects/490.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f490(const char *s) {
    int i = 0;
    while (i < strlen(s)) {
        assert(s[i] == '0' || s[i] == '1');
        i++;
    }
    assert(strcmp(s + strlen(s) - 3, "011") == 0);
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f490(my_string);
  return 0;
}
